app/lib/foreman_deployments/planner_visitor.rb

```diff
@@ -0,0 +1,26 @@
+module ForemanDeployments
+  class PlannerVisitor < BaseDereferenceVisitor
+    def initialize(parent_action)
+      super()
+      @parent_action = parent_action
+    end
+
+    protected
+
+    def visit_task_definition(task)
+      super
+      task.plan(@parent_action)
```

Comment:
I know about this case. You can't plan the task more than once. See https://github.com/theforeman/foreman_deployments/pull/6/files#diff-0c3747ebefa0b7a3186ef3ab2294f343R12 We keep the planned tasks cached on the task definition side.